dao/patient/etl-patient-dao.js

```diff
@@ -1,309 +1,354 @@
+      promise.then(function(result) {
         callback(result);
+      }).catch(function(err) {
+        callback(err);
       });
+    }
+    
+    // return the promise
+    return promise;
+  }
+  
+  function getClinicalNotes(request, callback) {
+    var patientEncounters = encounterService.getPatientEncounters(request.params.uuid);
+    var patientHivSummary = getPatientHivSummary(request);
+    var patientVitals = getPatientVitals(request);
+    
+        
+    Promise.all([patientEncounters, patientHivSummary, patientVitals]).then(function(data) {
+        var encounters = data[0];
+        var hivSummaries = data[1].result;
+        var vitals = data[2].result;
+        var notes = noteService.generateNotes(encounters,hivSummaries,vitals);
+        callback({notes:notes,status:'notes generated'});
+    })
+    .catch (function(e) {
+        // Return empty json on error
+        console.log('Error',e);
+        callback({notes:[],status:'error generating notes', error:e});
+    });
+  }
+  
+  function getPatientData(request, callback) {
+    var uuid = request.params.uuid;
+    var order =  helpers.getSortOrder(request.query.order);
 
-    },
-    getClinicalNotes: function getClinicalNotes(request, callback) {
-      var patientEncounters=noteService.getPatientEncounters(request);
-      var patientHivSummary=noteService.getHivSummary(request);
-      var patientVitals=noteService.getVitals(request);
+    var queryParts = {
+      columns: request.query.fields || "t1.*, t2.cur_arv_meds",
+      table: "etl.flat_labs_and_imaging",
+      joins: [
+        ['etl.flat_hiv_summary', 't2', 't1.encounter_id = t2.encounter_id']
+      ],
+      where: ["t1.uuid = ?", uuid],
+      order: order || [{
+        column: 'test_datetime',
+        asc: false
+      }],
+      offset: request.query.startIndex,
+      limit: request.query.limit
+    };
 
-      Promise.all([patientEncounters,patientHivSummary,patientVitals]).then(function(data) {
-          var encounters = data[0].results;
-          var hivSummaries = data[1].result;
-          var vitals = data[2].result;
-          noteService.generate(encounters,hivSummaries,vitals,function (notes) {
-              callback({notes:notes,status:'notes generated'});
-          });
+    db.queryServer_test(queryParts, function(result) {
 
-      })
-      .catch (function(e) {
-          // Return empty json on error
-          console.log('Error',e);
-          callback({notes:[],status:'error generating notes'});
+      _.each(result.result, function(row) {
+        row.tests_ordered = helpers.getTestsOrderedNames(row.tests_ordered);
+        row.cur_arv_meds = helpers.getARVNames(row.cur_arv_meds);
       });
-    },
-    getPatientData: function getPatientData(request, callback) {
-      var uuid = request.params.uuid;
-      var order =  helpers.getSortOrder(request.query.order);
+      callback(result);
+    });
+  }
+  
+  function getPatient(request, callback) {
```

Comment:
should the name of this be getPatient? Looks like it's getting flat_hiv_summary for a patient